File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set(nums.begin(), nums.end());
        int max_count = 0;
        if (nums.empty()) {
            return 0;
        }
        for (int i = 0; i < nums.size(); i++) {
            int count = 0;
            int x = nums[i];
            while (set.find(x+1) != set.end()) {
                x++;
                count++;
            }
            max_count = max(count, max_count);
        }
        return max_count + 1;
    }
};

File: Data Structures & Algorithms/house-robber/submission-5.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = [-1] * len(nums)
        def dfs(nums, i):
            if i >= len(nums):
                return 0
            if dp[i] != -1:
                return dp[i]
            dp[i] = max(dfs(nums, i + 1), dfs(nums, i + 2) + nums[i])
            return dp[i]
        return dfs(nums, 0)